utils/misc.ml

```diff
@@ -17,6 +17,13 @@ exception Fatal_error
 let fatal_error msg =
   prerr_string ">> Fatal error: "; prerr_endline msg; raise Fatal_error
 
+let fatal_errorf fmt =
+  let (_ : string) = Format.flush_str_formatter () in
+  Format.kfprintf (fun ppf ->
+      fatal_error (Format.flush_str_formatter ()))
+    Format.str_formatter
```

Comment:
> Does this matter given that the program is about to terminate?

Next year this code will be reused to provide another format printed here. It's fine if it is slower than it should (I did not try to share the error buffer in my implementation because the function is only called once), I'd rather avoid having it wronger than it should.